### PR TITLE
Clarification of ecql_reference exception for attribute id

### PR DESCRIPTION
**Pull request by petersmythe**
Update documentation to reflect Stack Exchange and email list findings

https://gis.stackexchange.com/a/475826/68995

Replaces https://github.com/geoserver/geoserver/pull/7414

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: doc/en/user/source/filter/ecql_reference.rst [hunk 1]
```diff
@@ -193,8 +193,9 @@ Attribute
 
 An attribute name denotes the value of a feature attribute.
 
-* Simple attribute names are sequences of letters and numbers,
-* Attribute names quoted with double-quotes may be any sequence of characters.
+* Simple attribute names are sequences of letters and numbers, e.g. `States123`
+* Attribute names quoted with double-quotes may be any sequence of characters, e.g. `"States!@#"`
+* `Note <https://gis.stackexchange.com/a/475826/68995>`_: `id` is a reserved word in the GML spec (for the FeatureID) and thus an attribute (or database column) named `id` must be quoted, e.g. `"id"`
```
Comment (aaime) [inline doc/en/user/source/filter/ecql_reference.rst:198]: There is some confusion here... "id" and "name" are indeed default attributes of the abstract GML feature (the base "class" for all the features), but the reason why you need to quote "id" in ECQL, is that it's a keyword in that language:

https://github.com/geotools/geotools/blob/main/modules/library/cql/src/main/jjtree/ECQLGrammar.jjt#L180

Comment (petersmythe) [reply doc/en/user/source/filter/ecql_reference.rst:198]: Ah, thank you.  I will update.

File: doc/en/user/source/filter/ecql_reference.rst [hunk 2]
```diff
@@ -193,8 +193,9 @@ Attribute
 
 An attribute name denotes the value of a feature attribute.
 
-* Simple attribute names are sequences of letters and numbers,
-* Attribute names quoted with double-quotes may be any sequence of characters.
+* Simple attribute names are sequences of letters and numbers, e.g. `States123`
+* Attribute names quoted with double-quotes may be any sequence of characters, e.g. `"States!@#"`
+* `Note <https://gis.stackexchange.com/a/475826/68995>`_: `id` is one of a few `reserved keywords <https://github.com/geotools/geotools/blob/main/modules/library/cql/src/main/jjtree/ECQLGrammar.jjt#L180>`_ in ECQL and thus an attribute (or database column) named `id` must be quoted, e.g. `"id"`
```
[the remark this answers is not in the document]
Comment (jodygarnett) [reply doc/en/user/source/filter/ecql_reference.rst:198]: We also try and have a markdown file to link to: https://github.com/geotools/geotools/blob/main/modules/library/cql/ECQL.md

File: doc/en/user/source/filter/ecql_reference.rst (hunk 1 above)
Comment (jodygarnett) [reply doc/en/user/source/filter/ecql_reference.rst]: Checking that reserved word is only used for deprecated syntax:

> <id predicate> ::= "ID" "IN" <id list> 

It does not seem to offer a lot of value, but I understand the difficulty in removing.

File: doc/en/user/source/filter/ecql_reference.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (petersmythe) [reply doc/en/user/source/filter/ecql_reference.rst]: Thanks Jody, I tried to link to your markdown file, but could not find a reference to the `id` keyword in it, so chose to stick to ECQLGrammar.jjt, but with a specific commit hash.

I might check that the 2 resources actually align properly, maybe one is out of date.

I trust that this is now described in an acceptable way.

[the remark this answers is not in the document]
Comment (jodygarnett) [reply doc/en/user/source/filter/ecql_reference.rst]: It does sure the markdown is out of date

**Pull request merged**